resources/metrics/profiling_shell.sh

```diff
@@ -1,45 +1,55 @@
 #!/bin/bash
 
 # this variant of the Profiling shell wrapper
-# uses only the Bash keyword `time`, and the generally
+# uses only the Bash keyword `time`, `perl` to computer
+# the time involved as miliseconds, and the generally
 # standard command `tee` to get run time. This was
 # needed because the BSD/sysv variants of `date` only
 # offer second-resolution. No milliseconds, no nano-
 # seconds.
 
-# Create copies of stdout & stderr's file descriptors
-# so that the output of the subshell that `time` runs
-# can be captured and used.
-exec 3>&1
-exec 4>&2
-
 # Write some log files, that can be post-processed
-# relatively easily.
-__stdout="${VANAGON_TARGET}.stdout.log"
-__stderr="${VANAGON_TARGET}.stderr.log"
+# relatively easily. While you're at it, store runtimes in batches.
+if [[ -n ${VANAGON_TARGET} ]]; then
+  __batchfile="${VANAGON_TARGET}.csv"
+  __stdout="${VANAGON_TARGET}.stdout.log"
+  __stderr="${VANAGON_TARGET}.stderr.log"
+else
+  # Alternatively, if there's no defined Target, just
+  # dump that stuff in the bitbucket
+  __batchfile="/dev/null"
+  __stdout="/dev/null"
+  __stderr="/dev/null"
+fi
 
 # Define a default time format (seconds and milliseconds),
 # and then use some insane output redirection hacks to capture
 # the output of the `time` keyword. We're using `tee` to
 # output to stdout & stderr but we should figure out how to
 # supress that for quieter Vanagon builds.
 TIMEFORMAT='%3R'
-__seconds="$( {
-  time bash -o pipefail "${@}" > >(tee -ai "${__stdout}" >&3) 2> >(tee -ai "${__stderr}" >&4 )
-} 2>&1)"
-
-# If any part of the pipeline failed, then
-# this status should correspond to whatever
-# failing status the pipeline returned. We
-# want to check & preserve that.
-if [[ ${PIPESTATUS[0]} -ne 0 ]]; then
-  exit "${PIPESTATUS[0]}"
-fi
+{ __seconds="$( {
+    time bash "${@}" > >(tee -ai "${__stdout}" >&3) 2> >(tee -ai "${__stderr}" >&4
+  ); } 2>&1 )";
+} 3>&1 4>&2
 
-# Finally, if everything completed successfully,
-# finesse the numbers and fire them off to our
-# statsd interface.
-if [[ "${VANAGON_TARGET}" ]] && [[ "${VANAGON_STATSD_HOST}" ]] && [[ "${VANAGON_STATSD_PORT}" ]]; then
-  __elapsed="$(echo "scale=0;${__seconds}*1000/1" | bc -l)"
-  echo -n "vanagon.${VANAGON_PROJECT}.${VANAGON_PLATFORM}.${VANAGON_TARGET}:${__elapsed}|ms" > "/dev/udp/${VANAGON_STATSD_HOST}/${VANAGON_STATSD_PORT}"
-fi
+# PIPESTATUS can be reset prety easily, so the value of this specific pipeline
+# should be preserved to enable correct post-processing.
+__status=("${PIPESTATUS[@]}")
+
+
+# allow a failure in the command pipeline to halt execution,
+# the same way Make would if the pipeline was being executed
+# directly in Bash without this wrapper. This will halt the job
+# without sending metrics for this job upstream.
+for value in "${__status[@]}"; do
+  if [[ ${value} -ne 0 ]]; then
+    exit "${value}"
+  fi
+done
+
+# Batch the runtime for this task
+echo "${__seconds}" >> "${__batchfile}"
```

Comment:
It's not, right now. I'm working on the code to slurp it back. Sending it from the remote target just wasn't working right and it was getting to the point where getting `profiling_shell.sh` to actually run on Windows was more important. I'm working on code to slurp these back for post-processing now.